libraries/aws_backend.rb

```diff
@@ -148,6 +148,19 @@ def initialize(opts)
       client_args[:client_args][:region] = opts[:aws_region] if opts[:aws_region]
       # below allows each resource to optionally and conveniently set an endpoint
       client_args[:client_args][:endpoint] = opts[:aws_endpoint] if opts[:aws_endpoint]
+      # below allows each resource to optionally and conveniently set max_retries
```

Comment:
Can we refactor this to something along the lines of the below?

```
# below allows each resource to optionally and conveniently set max_retries
ENV.each do |key, val|
  opts[:retry_limit]=val.to_i if key.downcase == 'aws_retry_limit'
  opts[:retry_backoff]=val if key.downcase == 'aws_retry_backoff'
end

client_args[:client_args][:retry_limit] = opts[:aws_retry_limit] if opts[:retry_limit]
client_args[:client_args][:retry_backoff] = "lambda { |c| sleep(#{opts[:aws_retry_backoff]}) }" if opts[:retry_backoff]
```

Reply:
Edited the above to not override from the environment if the option is provided.

Reply:
how about instead of looping over ENV we do something along these lines:
```
env_hash=ENV.to_h.transform_keys(&:downcase)
opts[:retry_limit]=env_hash['retry_limit'].to_i if !opts[:retry_limit]
opts[:retry_backoff]=env_hash['retry_backoff'].to_i if !opts[:retry_backoff]
```

Reply:
looks good 👍